src/ensembl/src/content/app/entity-viewer/gene-view/components/default-transcripts-list/DefaultTranscriptsList.tsx

```diff
@@ -14,8 +19,54 @@ type Props = {
   rulerTicks: TicksAndScale;
 };
 
+const compareTranscriptLength = (
+  transcriptOne: Transcript,
+  transcriptTwo: Transcript
+) => {
+  const transcriptOneLength = getFeatureLength(transcriptOne);
+  const transcriptTwoLength = getFeatureLength(transcriptTwo);
+
+  if (transcriptOneLength < transcriptTwoLength) {
+    return -1;
+  }
+
+  if (transcriptOneLength > transcriptTwoLength) {
+    return 1;
+  }
+
+  return 0;
+};
+
+const sortTranscripts = (transcripts: Transcript[]) => {
+  const transcriptsWithCds = transcripts
+    .filter((transcript) => transcript.cds)
+    .sort(compareTranscriptLength);
+
+  const transcriptsWithoutCds = transcripts.filter(
+    (transcript) => !transcript.cds
+  );
+
+  const proteinCodingTranscripts = transcriptsWithoutCds
+    .filter((transcript) => transcript.biotype === 'protein_coding')
+    .sort(compareTranscriptLength);
+
+  const nonProteinCodingTranscripts = sortBy(
+    transcriptsWithoutCds.filter(
+      (transcript) => transcript.biotype !== 'protein_coding'
+    ),
+    ['biotype']
+  );
```

Comment:
is this a sort in alphabetic order? is this what Andrea wants?

Reply:
Yes, sorting based on the alphabetical order of the biotype for the non protein coding transcripts. This is criteria number 3.